gateway/1.5/modules/ROOT/pages/policies-pdk-configure-features.adoc

```diff
@@ -1,5 +1,589 @@
+
+The code examples assume you have some prior knowledge of the Rust Programming Language. For more information about programming in Rust, see:
+
+* https://doc.rust-lang.org/book/[The Rust Programming Language Documentation ^]
+* https://rust-lang-nursery.github.io/rust-cookbook/intro.html[Rust Cookbook ^]
+
+
+== Before You Begin
+
+* xref:policies-pdk-create-project.adoc[Create a New Project]
+* xref:policies-create-schema-definition.adoc[Define a Schema Definition]
++
+NOTE: You may need to add or redefine configuration parameters while implementing your policy. 
+
+* Review <<policy-template>> before adding additional features.
+
+== Policy Template
+
+PDK provides the initial `src/lib.rs` file in the policy project as a template to begin impletmenting your policy:
+
+[source,Rust]
+----
+// Copyright 2023 Salesforce, Inc. All rights reserved.
+mod generated;
+
+use anyhow::Result;
+
+use pdk::api::hl::*;
+
+use crate::generated::config::Config;
+
+// This filter shows how to log a specific request header.
+// You can extend the function and use the configurations exposed in config.rs file
+async fn request_filter(request_state: RequestState, _config: &Config) {
+    let headers_state = request_state.into_headers_state().await;
+    let token = headers_state.handler().header("Token").unwrap_or_default();
+    // Log the header value
+    logger::info!("Header value: {token}");
+}
+
+#[entrypoint]
+async fn configure(launcher: Launcher, Configuration(bytes): Configuration) -> Result<()> {
+    let config = serde_json::from_slice(&bytes)?;
+    let filter = on_request(|rs| request_filter(rs, &config));
+    launcher.launch(filter).await?;
+    Ok(())
+}
+----
+
+The policy performs some simple logging. For each incoming request, the policy logs the header `token` at the `info` log level. 
+
+The following elements configure the policy:
+
+* `use pdk::api::hl::*;`: Imports all the necessary components of the PDK into the `lib.rs` source code.
+* `#[entrypoint]` function: Executes when the policy is applied and calls the `request_filter` function. Variables defined inside this function are avaliable while the policy is applied.
++
+The `#[entrypoint]` function receives the following parameters:
++
+** `Launcher`: Sets the filter functions.
+** `Configuration`: Provides the configuration parameters defined in xref:policies-pdk-create-schema-definition.adoc[]. 
+
+* `request_filter`: Executes once per every request sent to the API instance to which the policy is applied. This function is the implementation of the filtering performed by the policy. Variables defined inside this function available for the duration of the request.
++
+The `request_filter` is an example of a wrapped function
++
+The `#[entrypoint]` function executes wrapped functions:
++
+[source,Rust]
+----
+let filter = on_request(|rs| request_filter(rs, &config));
+launcher.launch(filter).await?;
+----
+
```

Comment:
Is it possible to add a sub-section here? Otherwise, the example is too long to follow. Something like: 
=== Processing the Outgoing Response
=== Using an on_response Wrapper
Not sure if these titles are accurate, but something like this would help understand that this is some additional information to the first policy template.